src/platform/backends/qemu/linux/firewall_config.cpp

```diff
@@ -336,7 +338,10 @@ mp::FirewallConfig::FirewallConfig(const QString& bridge_name, const std::string
 
 mp::FirewallConfig::~FirewallConfig()
 {
-    mp::top_catch_all(category, [this] { clear_all_firewall_rules(); });
+    if (!firewall.isEmpty())
+    {
+        mp::top_catch_all(category, [this] { clear_all_firewall_rules(); });
+    }
```

Comment:
Yes, since the `clang-format` showed uncovered changes, I wrote some unit tests to cover that, but found in the testing that this is an issue and is fixed by adding this.